### ISSUE TITLE
[BUG] base-2016 template index.twig

### ISSUE THREAD

**Issue by mborn319** [bolt/bolt#5446]
Attempting to edit the template for base-2016/index.twig by following the link from the frontend throws an error. because (firstly) the .twig file does not exist, and secondly, the catch expects `$file` to be an object when it is only a string.

So, the "Edit this template" button goes to `http://brinks.localhost/bolt/file/edit/themes/foundation/index.twig`, when it SHOULD point to `http://brinks.localhost/bolt/file/edit/themes/base-2016/index.twig`.

Details
-------

 - Relevant Bolt Version: 3.0.4
 - Install type: Download from https://bolt.cm/distribution/bolt-latest.tar.gz
 - PHP version: 7.0
 - Used webserver: Apache

Reproduction
------------

 1. **Bug summary**: On "homepage" view, clicking "Edit this template" in the "Welcome to the Bolt Foundation theme. " callout throws a `Call to a member function getPath() on string` error. 

 2. **Specifics**: This is a new installation of Bolt CMS 3.0.4. I am using the default routes and the `base-2016` template.
I have commented out the "Showcases" contenttype and all taxonomies.
I have added custom database info and modified menu.yml.
Other than that, this is a vanilla install.

 3. **Steps to reproduce**: 
* Install Bolt CMS, add a "homepage" page with the index.twig template. 
* Preview the template. 
* Click "Edit this template"
 4. **Expected result**: I expected to get a text editor to edit the Twig template for the homepage.
 5. **Actual result**: I received a page with a PHP error message: `Call to a member function getPath() on string in /vendor/bolt/bolt/src/Controller/Backend/FileManager.php:86`.

Screenshot: ![bolt-cms-php-error](https://cloud.githubusercontent.com/assets/8106227/16028792/c64be55c-31b0-11e6-9bc2-784be62543dc.png)

![bolt-cms-edit-template-link](https://cloud.githubusercontent.com/assets/8106227/16029033/6d1bf58c-31b3-11e6-87d0-dbfae7284cb0.png)

**Comment by mborn319** [comment]
Okay, so I found that this was just fixed in https://github.com/bolt/bolt/commit/8432807bcb5f0c7cddcc443bb70ebe4f6d4235bb. 

However, I think that `src/Controller/Backend/FileManager.php` should be updated so that it won't fail if this file doesn't exist. That's the WHOLE POINT of the `try...catch` block.

````php
 80         try {
 81             /** @var File $file */
 82             $file = $filesystem->get($file);
 83             $type = Lib::getExtension($file->getPath());
 84             $data = ['contents' => $file->read()];
 85         } catch (FileNotFoundException $e) {
 86             $error = Trans::__('general.phrase.file-not-exist', ['%s' => $file->getPath()]);
 87             $this->abort(Response::HTTP_NOT_FOUND, $error);
 88         } catch (IOException $e) {
 89             $error = Trans::__('general.phrase.file-not-readable', ['%s' => $file->getPath()]);
 90             $this->abort(Response::HTTP_NOT_FOUND, $error);
 91         }
````

Feel free to discuss or close.

**Comment by GawainLynch** [comment]
@mborn319 Yeah, unfortunately the version of `bolt/filesystem` landed late in 3.0-dev due to … time. :wink: 

But yes, a more user friendly approach is 100% the desired outcome!
